### ISSUE TITLE
Add `--wait` to commands where this flag is missing

### ISSUE THREAD

**Issue by nikimanoledaki** [weaveworks/eksctl#5385]
### Description

Following up from #5250, the aim of this ticket is to find commands that would benefit from `--wait`  but are missing this flag, and add the flag to these commands.

For example, some suggestions: 
- `eksctl drain nodegroup`
~- `eksctl scale nodegroup`~ Ticket #6396
- `eksctl deregister/register cluster`

**Comment by arturzeitler** [comment]
I'd would like to take on this issue. 
I noticed that drain and scale already have some wait functionality but unlike the changes from update [#5250](https://github.com/weaveworks/eksctl/issues/5250) they obviously do not have a wait flag and their wait functionality is not linked to the waiter package from eksctl.
Apart from adding the wait flag, would the goal be to extend their current wait functionality with the native waiter from eksctl?

**Comment by Himangini** [comment]
As part of this ticket we'll add `--wait` flag for the following command
- `eksctl drain nodegroup`

We'll open a new ticket for the following commands 
- `eksctl register cluster`
- `eksctl deregister/register cluster`

**Comment by TiberiuGC** [comment]
After further investigation it turns out that `eksctl scale nodegroup` is the single command out of the ones mentioned in this ticket that can really benefit from supporting the `--wait` flag. As for the rest: 
- `eksctl drain nodegroup` - this command already waits by default for pods to be evicted from the nodes https://github.com/weaveworks/eksctl/blob/f261471677c75703d0da0d643eb3c1ed268ca3d4/pkg/drain/nodegroup.go#L205-L247
- `eksctl register / deregister cluster` - the EKS API calls used for these commands don't trigger any async process. So there's nothing to wait on.
